hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperTask.java

```diff
@@ -252,48 +253,54 @@ private void handleDeleteOMKeyEvent(String key,
                                       List<ContainerKeyPrefix>
                                           deletedContainerKeyList)
       throws IOException {
+
     Set<ContainerKeyPrefix> keysToBeDeleted = new HashSet<>();
-    try (TableIterator<ContainerKeyPrefix,
-        ? extends Table.KeyValue<ContainerKeyPrefix, Integer>> containerIterator
-             = reconContainerMetadataManager.getContainerTableIterator()) {
+    try (TableIterator<KeyPrefixContainer, ? extends
+        Table.KeyValue<KeyPrefixContainer, Integer>> keyContainerIterator =
+             reconContainerMetadataManager.getKeyContainerTableIterator()) {
 
       // Check if we have keys in this container in the DB
-      while (containerIterator.hasNext()) {
-        Table.KeyValue<ContainerKeyPrefix, Integer> keyValue =
-            containerIterator.next();
+      keyContainerIterator.seek(new KeyPrefixContainer(key));
+      while (keyContainerIterator.hasNext()) {
+        Table.KeyValue<KeyPrefixContainer, Integer> keyValue =
+            keyContainerIterator.next();
         String keyPrefix = keyValue.getKey().getKeyPrefix();
         if (keyPrefix.equals(key)) {
-          keysToBeDeleted.add(keyValue.getKey());
+          if (keyValue.getKey().getContainerId() != -1) {
+            keysToBeDeleted.add(keyValue.getKey().toContainerKeyPrefix());
+          }
+        } else {
+          break;
         }
       }
-    }
```

Comment:
Hmm, why not release `keyContainerIterator` here?
Do we have other reasons to keep `keyContainerIterator` til the method end?

Reply:
Hi @symious , other changes looks good now. Can we move the closing bracket (for the `try`) back here?